### PR TITLE
add generated table of contents

### PR DESCRIPTION
**Pull request by ajrussellaudio**
## What?

Adds automatically generated tables of contents to the Community and Learning Resources pages

## Why?

Because we can. Also it means sections can be added without having to also update the manually built ToC.

## Wow, really?

Yep. Markdown translation in Jekyll is handled by [kramdown](https://kramdown.gettalong.org/), which already generates selector IDs for headings. That's how we get anchor links like this: https://codeclanalumni.github.io/survival-guide/pages/learning-resources#javascript-online-resources I haven't added any new libraries here. Kramdown can just take all these generated IDs and [put them all in a list](https://kramdown.gettalong.org/converter/html.html#toc).

 tl;dr:

```
# Title
{:.no_toc}
<!-- We don't want the page title to appear in the table of contents -->

- TOC
{:toc}
<!-- We want the table of contents to be an unordered list, so we make an unordered list of one item (the text here doesn't matter) -->
<!-- kramdown will replace the unordered list on the line above the {:toc} tag with a generated TOC -->

## Another thing

## And another thing

Actual text content

### Not only that, but also
```

Screenshot of generated table of contents

Pretty sick, right?

### CODE REVIEW

**Review by artiepreston — APPROVED** [review]
Looks good 👍

**Comment by ajrussellaudio** [comment]
Cheers @ktweeden :+1:
